dask/dataframe/io/parquet/arrow.py

```diff
@@ -317,6 +318,36 @@ def _need_fragments(filters, partition_keys):
     return bool(filtered_cols - partition_cols)
 
 
+def _hive_dirname(name, val):
+    # Simple utility to produce hive directory name
+    val = "__HIVE_DEFAULT_PARTITION__" if pd.isna(val) else val
+    return f"{name}={val}"
+
+
+def _process_kwargs(partitioning=None, warn=False, **kwargs):
+    # Pre-process a dict of `pyarrow.dataset.dataset`` key-word
+    # arguments. Primary purpose is to convert a dictionary-based
+    # "partitioning" option into a proper `Partitioning` object
+    if warn and not isinstance(partitioning, (str, list, dict)):
+        # Optionally warn the user if their partitioning
+        # arugument is likely to cause failures in distributed
+        warnings.warn(
+            f"Beware that this `partitioning` argument "
+            f"(type {type(partitioning)}) may not be serializable, "
+            f"and may lead to errors in a distributed cluster. "
+            f"Please pass in a dictionary of key-word arguments for "
+            f"`pyarrow.dataset.partitioning` to avoid this problem."
+        )
```

Comment:
This feels a bit premature. I'm not sure what values are currently supported for the `"partitioning"` key, but maybe documentation is sufficient? It also doesn't apply to the threaded scheduler (though I know the warning says "_may_ not be serializable")

Reply:
I removed the warning, because I agree it is a bit much. However, I'm a bit unsure of the best place to document the option to pass through a dictionary of key-word arguments.

Reply:
Update: Added a bit of info about this to the doc-string under the `dataset` param.